src/pages/SearchPage.tsx

```diff
@@ -12,42 +12,39 @@ import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import Performance from '@libs/Performance';
 import * as ReportUtils from '@libs/ReportUtils';
+import type {OptionData} from '@libs/ReportUtils';
 import * as Report from '@userActions/Report';
 import Timing from '@userActions/Timing';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import personalDetailsPropType from './personalDetailsPropType';
-import reportPropTypes from './reportPropTypes';
-
-const propTypes = {
-    /* Onyx Props */
+import type * as OnyxTypes from '@src/types/onyx';
 
+type SearchPageProps = {
     /** Beta features list */
-    betas: PropTypes.arrayOf(PropTypes.string),
+    betas: OnyxEntry<OnyxTypes.Beta[]>;
 
     /** All of the personal details for everyone */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
+    personalDetails: OnyxEntry<OnyxTypes.PersonalDetailsList>;
 
-    /** All reports shared with the user */
-    reports: PropTypes.objectOf(reportPropTypes),
+    /** The report currently being looked at */
+    reports: OnyxEntry<OnyxTypes.Report>;
 
     /** Whether we are searching for reports in the server */
-    isSearchingForReports: PropTypes.bool,
+    isSearchingForReports: OnyxEntry<boolean>;
 };
```

Comment:
This way it will be easier add props if needed in the future and the structure will be the same as in other places
```suggestion
};

type SearchPageProps = SearchPageOnyxProps;
```

Reply:
I've just realized it's a page, so it should have navigation props, so it can look like this
```suggestion
};

type SearchPageProps = SearchPageOnyxProps & StackScreenProps<SearchNavigatorParamList, typeof SCREENS.SEARCH_ROOT>;
```

Reply:
@ruben-rebelo how about this comment?